src/asynchronous/server.rs

```diff
@@ -327,13 +327,14 @@ impl Builder for ServerBuilder {
                 server_shutdown: self.shutdown_waiter.clone(),
                 handler_shutdown: disconnect_notifier,
             },
-            ServerWriter { rx },
+            ServerWriter { rx, _server_shutdown: self.shutdown_waiter.clone() },
         )
     }
 }
 
 struct ServerWriter {
     rx: MessageReceiver,
+    _server_shutdown: shutdown::Waiter
```

Comment:
don't need to call `wait` in `exit`?

Reply:
There is no need for that, the _server_shutdown waiter acts as a barrier for the shutdown.wait_all_exit() call found here: https://github.com/containerd/ttrpc-rust/blob/master/src/asynchronous/server.rs#L243-L254 

When the ServerWriter goes out of scope [here](https://github.com/containerd/ttrpc-rust/blob/master/src/asynchronous/connection.rs#L68-L69), after sending all replies, the _server_shutdown waiter is dropped. When that happens the server shutdown notifier will return from the wait_all_exit() call.